force-app/main/default/classes/example/AccountDomainTest.cls

```diff
@@ -197,19 +210,133 @@ public with sharing class AccountDomainTest {
 
     //verify
     mockService.assertThat().method(mockedMethodName).wasCalled(1);
-    System.assert(!hasException, 'Mocked exception was not thrown');
-    Account acct = (Account) mockService.forMethod('doInsert').getValueOf('acct');
-    System.assertNotEquals(null, acct.Id, 'Account Id is null after insert');
+    System.assert(!hasException, MOCKED_EXCEPTION_WAS_NOT_THROWN);
+    Account acct = (Account) mockService.forMethod(DO_INSERT).getValueOf(ACCT);
+    System.assertNotEquals(null, acct.Id, ACCOUNT_ID_IS_NULL_AFTER_INSERT);
+  }
+
+  @IsTest
+  private static void create_two_accounts_should_succeed_if_both_are_successful() {
+    //setup
+    String mockedMethodName = DO_INSERT;
+    Database.SaveResult mockedSaveResult = generateSaveResult(true);
+    mockService.when(mockedMethodName).thenReturn(mockedSaveResult);
+    AccountDomain.AccountDomainException myException;
+
+    //test
+    Test.startTest();
+    try {
+      sut.createTwoPublicAccounts(ACME);
+    } catch (AccountDomain.AccountDomainException e) {
+      myException = e;
+    }
+    Test.stopTest();
+
+    //verify
+    System.assertEquals(null, myException, 'An exception should not be thrown because both of the Database.SaveResult were successful');
```

Comment:
Would be useful to also print out what the exception was to help with debugging efforts.